ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -282,26 +275,26 @@ func ipfsErrorResponder(w http.ResponseWriter, errMsg string) {
 }
 
 func (ipfs *Connector) pinOpHandler(op string, w http.ResponseWriter, r *http.Request) {
-	argA := r.URL.Query()["arg"]
-	if len(argA) == 0 {
+	arg, ok := extractArgument(r.URL)
+	if !ok {
 		ipfsErrorResponder(w, "Error: bad argument")
 		return
 	}
-	arg := argA[0]
 	_, err := cid.Decode(arg)
 	if err != nil {
 		ipfsErrorResponder(w, "Error parsing CID: "+err.Error())
```

Comment:
@hsanjuan So for some reason, the tests that attempt to pin/unpin a bad cid are failing on travis/jenkins (they aren't failing locally and I have tried on two separate computers/OSes). It would appear that on travis/jenkins, this `cid.Decode` operation is returning an error when passed the `test.ErrorCid`